lib/game.rb

```diff
@@ -1,17 +1,84 @@
 # frozen_string_literal: true
 
-# class Game checks for the rows,columns and diagonals to check if anyone has won or else
-# tracks the count of turns and returns a draw if the bosrd is full before anyone wins.
+require_relative '../lib/board.rb'
+require_relative '../lib/ui.rb'
+
+# class Game Initiate a new board (using Board class) and status,
+# plays one match, and update the UI (using Ui module) and change
+# the player score in the mean time.
 class Game
-  def initialize
-    @count = 0
+  include Ui
+
+  def initialize(player1, player2)
+    @player1 = player1
+    @player2 = player2
+    new_match
+  end
+
+  private
+
+  def new_match
+    @player1.add_mark(nil)
+    @status = 'continue'
+    @rounds = 0
+    @board = Board.new
+    @marked_cells = []
+    @turn = @player2
+    assign_players_signs
+    play
+  end
+
+  def assign_players_signs
+    while @player1.mark.nil?
+      user_input = prompt("\nPlayer 1, Please choose a mark? (X/Y) ").upcase
+      break if %w[X O].include?(user_input)
```

Comment:
You ask for the user to enter either **X** or **Y** but you break only if he entered **X** or **O**a

Reply:
Thanks, I will solve this in the develop branch after merging it to the master!